operator/src/main/java/oracle/kubernetes/weblogic/domain/model/DomainStatus.java

```diff
@@ -116,28 +117,24 @@ public DomainStatus(DomainStatus that) {
    */
   public DomainStatus addCondition(DomainCondition newCondition) {
     if (conditions.contains(newCondition)) {
-      conditions = conditions.stream()
-          .filter(c -> preserve(c, newCondition.getType().typesToRemoveAlways())).collect(Collectors.toList());
       return this;
     }
 
-    conditions = conditions.stream()
-        .filter(c -> preserve(c, newCondition.getType().typesToRemove())).collect(Collectors.toList());
+    conditions = conditions.stream().filter(c -> preserve(c, newCondition.getType())).collect(Collectors.toList());
 
     conditions.add(newCondition);
     reason = newCondition.getStatusReason();
     message = newCondition.getStatusMessage();
     return this;
   }
 
-  private boolean preserve(DomainCondition condition, DomainConditionType[] types) {
-    for (DomainConditionType type : types) {
-      if (condition.getType() == type) {
-        return false;
-      }
+  // Returns true if adding a condition of the new type should not remove the specified condition.
+  private boolean preserve(DomainCondition condition, DomainConditionType newType) {
+    if (newType == condition.getType() && !"True".equalsIgnoreCase(condition.getStatus())) {
```

Comment:
this condition will never be true because the code returns in line 120 when newType already exists.

Reply:
That code should only return if the conditions are equal. This tests whether _another_ condition of the same type exists and has status True.